core/iwasm/common/wasm_runtime_common.c

```diff
@@ -712,6 +739,34 @@ get_package_type(const uint8 *buf, uint32 size)
     return Package_Type_Unknown;
 }
 
+#if WASM_ENABLE_LIBC_WASI != 0 || WASM_ENABLE_MULTI_MODULE != 0
+#define find_export(module, module_name, field_name, export_kind, error_buf, \
```

Comment:
here use macro is to avoid the redundent definition, if we define it as a function, we have to introduce WASMModuleCommon *module back in the function, in which using #if WASM_ENABLE_AOT != 0 and #if WASM_ENABLE_INTERP != 0 for different cases. 
and actually, the macro find_export has only one caller:  loader_find_export, who will exit also when find_export returns false.

Reply:
Maybe you can define the function as `bool find_export(WASMExport *exports, uint32 export_count, const char *module_name, const char *field_name, ...)`, and pass module->exports and module->export_count to it?

Reply:
Note that AOTExport is same as WASMExport (see aot.h), how about directly implementing all the code in loader_find_export:
```C
WASMExport *
loader_find_export(const WASMModuleCommon *module, const char *module_name,
                   const char *field_name, uint8 export_kind, char *error_buf,
                   uint32 error_buf_size)
{
    WASMExport *exports = NULL, *export;
    uint32 export_count = 0, i;

#if WASM_ENABLE_INTERP != 0
    if (module->module_type == Wasm_Module_Bytecode) {
        WAMSModule *wasm_module = (WASMModule *)module;
        exports = wasm_module->exports;
        export_count = wasm_module->export_count;
    }
#endif
#if WASM_ENABLE_AOT != 0
    if (module->module_type == Wasm_Module_AoT) {
        AOTModule *aot_module = (AOTModule *)module;
        exports = aot_module->exports;
        export_count = aot_module->export_count;
    }
#endif

    for (i = 0, export = exports; i < export_count; ++i, ++export) {
       ...
    }

exit:
    return export;
}
```